vulkan/vulkan_device.cc

```diff
@@ -273,14 +274,40 @@ bool VulkanDevice::ChooseSurfaceFormat(const VulkanSurface& surface,
     return false;
   }
 
+  std::map<VkFormat, VkSurfaceFormatKHR> supported_formats;
+
   for (uint32_t i = 0; i < format_count; i++) {
     if (GrVkFormatToPixelConfig(formats[i].format, nullptr /* dont care */)) {
-      *format = formats[i];
+      supported_formats[formats[i].format] = formats[i];
+    }
+  }
+
+  if (supported_formats.size() == 0) {
+    return false;
+  }
+
+  const std::vector<VkFormat> desired_formats = {
+      VK_FORMAT_R8G8B8A8_SRGB,       // kSRGBA_8888_GrPixelConfig
+      VK_FORMAT_B8G8R8A8_SRGB,       // kSBGRA_8888_GrPixelConfig
+      VK_FORMAT_R8G8B8A8_UNORM,      // kRGBA_8888_GrPixelConfig
+      VK_FORMAT_B8G8R8A8_UNORM,      // kBGRA_8888_GrPixelConfig
+      VK_FORMAT_R16G16B16A16_SFLOAT  // kRGBA_half_GrPixelConfig
```

Comment:
My only suggestion would be to prioritize this over the non-sRGB configs (move it up two spots in the list). That will ensure that you get linear blending and color correctness if it's at all possible.